hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/lock/ConditionalWriteLockConfigTest.java

```diff
@@ -0,0 +1,112 @@
+    props.setProperty(ConditionalWriteLockConfig.HEARTBEAT_POLL_MS.key(), "10000");
+    props.setProperty(BASE_PATH.key(), "/hudi/table/basepath");
+
+    ConditionalWriteLockConfig config = new ConditionalWriteLockConfig.Builder()
+        .fromProperties(props)
+        .build();
+
+    assertEquals("s3://bucket/path/locks", config.getLocksLocation());
+    assertEquals(120000, config.getLockValidityTimeoutMs());
+    assertEquals(10000, config.getHeartbeatPollMs());
+    assertEquals("/hudi/table/basepath", config.getHudiTableBasePath());
+  }
+
+  @Test
+  void testMissingRequiredProperties() {
+    TypedProperties props = new TypedProperties();
+    props.setProperty(ConditionalWriteLockConfig.LOCK_VALIDITY_TIMEOUT_MS.key(), "120000");
+    ConditionalWriteLockConfig.Builder propsBuilder = new ConditionalWriteLockConfig.Builder();
+
+    IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+        () -> propsBuilder.fromProperties(props));
+    assertTrue(exception.getMessage().contains(BASE_PATH.key()));
+    props.setProperty(BASE_PATH.key(), "s3://bucket/path/locks");
+    props.setProperty(ConditionalWriteLockConfig.LOCK_INTERNAL_STORAGE_LOCATION.key(), "s3://bucket/path/locks");
+    assertThrows(IllegalArgumentException.class, () -> propsBuilder.fromProperties(props));
+  }
+
+  @Test
+  void testTimeThresholds() {
+    TypedProperties props = new TypedProperties();
+    props.setProperty(BASE_PATH.key(), "/hudi/table/basepath");
+    props.setProperty(ConditionalWriteLockConfig.LOCK_VALIDITY_TIMEOUT_MS.key(), "5000");
+    props.setProperty(ConditionalWriteLockConfig.HEARTBEAT_POLL_MS.key(), "3000");
+    ConditionalWriteLockConfig.Builder propsBuilder = new ConditionalWriteLockConfig.Builder();
+
+    IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+        () -> propsBuilder.fromProperties(props));
+    assertTrue(exception.getMessage().contains(ConditionalWriteLockConfig.LOCK_VALIDITY_TIMEOUT_MS.key()));
+    props.setProperty(ConditionalWriteLockConfig.LOCK_VALIDITY_TIMEOUT_MS.key(), "4999");
+    props.setProperty(ConditionalWriteLockConfig.HEARTBEAT_POLL_MS.key(), "1000");
+    exception = assertThrows(IllegalArgumentException.class, () -> propsBuilder.fromProperties(props));
+    assertTrue(exception.getMessage().contains(ConditionalWriteLockConfig.LOCK_VALIDITY_TIMEOUT_MS.key()));
+    props.setProperty(ConditionalWriteLockConfig.LOCK_VALIDITY_TIMEOUT_MS.key(), "5000");
+    props.setProperty(ConditionalWriteLockConfig.HEARTBEAT_POLL_MS.key(), "999");
+    exception = assertThrows(IllegalArgumentException.class, () -> propsBuilder.fromProperties(props));
+    assertTrue(exception.getMessage().contains(ConditionalWriteLockConfig.HEARTBEAT_POLL_MS.key()));
+  }
+
+  @Test
+  void testBucketPathValidation() {
+    TypedProperties props = new TypedProperties();
+    props.setProperty(ConditionalWriteLockConfig.LOCK_INTERNAL_STORAGE_LOCATION.key(), "invalid/path");
```

Comment:
Add another test on the config value of `<table_base_path>/.hoodie` and `<table_base_path>/partition` so both of them should fail the config check.